tools/ldcontext_generator.py

```diff
@@ -1,23 +1,53 @@
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
 """
-Extracts the properties, types and  enumerations of a JSON Schema
-converting them into terms of a JSON-LD @Context
+This script provides two files:
+ - context.jsonld, that serves https://schema.lab.fiware.org/ld/fiware-datamodels-context.jsonld
+ - mapping_list.yml, that serves  https://uri.fiware.org/ns/datamodels
+
+context.jsonld is combined by extracting the properties, types and  enumerations of a JSON Schema and
+converting them into terms of a JSON-LD @Context. mapping_list.yml uses the result of extracting
+to prepare a list of terms with schemas and specifications.
 
 Copyright (c) 2019 FIWARE Foundation e.V.
 
 Author: José M. Cantera
 """
 
-import sys
 import json
+import yaml
 import os
 from datetime import datetime
+from argparse import ArgumentParser
 
 # Here the aggregated @context will be stored
 aggregated_context = {
 }
 
+# Here a list of mappings (attribute -> schema/spec) will be stored
+mapping_list = {
+    "terms": {}
+}
+
+# Here a list of spec alerts will be stored
+alert_list = [
+]
+
+# Template to prepare a valid URL of a schema for a mapping
+schema_url = 'https://github.com/FIWARE/dataModels/blob/master/{}'
+
+# Agri* schemas stores at another github organization
+agri_url = 'https://github.com/GSMADeveloper/NGSI-LD-Entities/blob/master/definitions/{}.md'
+agri_mapping = {
```

Comment:
can we externalize these mappings into a file?

Reply:
Done